js/views/transactionsVw.js

```diff
@@ -281,6 +285,118 @@ module.exports = baseVw.extend({
     tabWrapper.append(orderShort.render().el);
   },
 
+  clickDownloadCSV: function(e){
+    var targBtn = $(e.target);
+    targBtn.closest('.btn').addClass('loading');
+    this.downloadCSV(targBtn);
+  },
+
+  downloadCSV: function(targBtn){
+    var self = this,
+        rawData = [],
+        calls = [],
+        dataCSV = '',
+        tempAnchor = document.createElement('a'),
+        dataBlob,
+        exportData = function(data){
+          "use strict";
+          dataCSV = Papa.unparse(data);
+          dataBlob = new Blob([dataCSV], {'type':'application\/octet-stream'});
+          tempAnchor.href = window.URL.createObjectURL(dataBlob);
+          tempAnchor.download = 'export.csv';
+          tempAnchor.click();
+        };
+
+    //clear existing data
+    this.currentExportData = [];
+
+    switch(this.state){
+      case "purchases":
+        rawData = this.purchasesCol.toJSON();
+        break;
+      case "sales":
+        rawData = this.salesCol.toJSON();
+        break;
+      case "cases":
+        rawData = this.casesCol.toJSON();
+        break;
+    }
+
+    $.each(rawData,(i, transaction)=> {
+      transaction.status = polyglot.t('transactions.OrderStatus'+transaction.status);
+      transaction.timestamp = new Date(transaction.timestamp * 1000);
+      transaction = __.omit(transaction, "thumbnail_hash", "btAve", "imageUrl", "order_date", "cCode", "displayPrice", "vendor", "transactionType");
+      calls.push(this.getTransactionData(transaction.order_id, transaction));
+    });
+
+    $.when.apply(null, calls)
+        .done(function(){
+          exportData(self.currentExportData);
+          targBtn && targBtn.removeClass('loading');
+        });
+  },
+
+  getTransactionData: function(orderID, dataObject){
+    dataObject = dataObject || {};
+
+    var getCall = $.getJSON(this.serverUrl + 'get_order',{'order_id': orderID}, function(data){
```

Comment:
Re: API, we don't have changes to the backend in the scope of this task, and in my testing this has been pretty quick even for large numbers of calls.

If it gets unmanageable, we may have to loop back and do something on the backend.